overlord/devicestate/firstboot.go

```diff
@@ -415,10 +418,30 @@ var loadDeviceSeed = func(st *state.State, sysLabel string) (deviceSeed seed.See
 		return assertstate.AddBatch(st, batch, nil)
 	}
 
-	if err := deviceSeed.LoadAssertions(assertstate.DB(st), commitTo); err != nil {
+	db := assertstate.DB(st)
+	if err := deviceSeed.LoadAssertions(db, commitTo); err != nil {
 		return nil, err
 	}
 
+	// if model is dangerous, check if there is auto-import.assert and import it
+	if deviceSeed.Model().Grade() == asserts.ModelDangerous {
+		seed20AssertionsLoader, ok := deviceSeed.(seed.AutoImportAssertionsLoaderSeed)
+		if !ok {
+			logger.Noticef("failed to cast seed20 assertion loader")
```

Comment:
I'm unsure of this error message if it's maybe a bit too specific? Maybe something more describtive as `failed to auto-import assertion, invalid assertion type`

Reply:
Fair point, I have changed it to: "failed to auto-import assertions, invalid loader"